test/rpktest.c

```diff
@@ -0,0 +1,759 @@
+    unsigned char cert_type_bad[] = { 0xFF };
+    unsigned char cert_type_extra[] = { TLSEXT_cert_type_rpk,
+                                        TLSEXT_cert_type_x509,
+                                        0xFF };
+    unsigned char cert_type_unsup[] = { TLSEXT_cert_type_pgp,
+                                        TLSEXT_cert_type_1609dot2 };
+    unsigned char cert_type_just_x509[] = { TLSEXT_cert_type_x509 };
+    unsigned char cert_type_just_rpk[] = { TLSEXT_cert_type_rpk };
+
+    if (!TEST_true(create_ssl_ctx_pair(NULL,
+                                       TLS_server_method(), TLS_client_method(),
+                                       TLS1_2_VERSION, TLS1_2_VERSION,
+                                       &sctx, &cctx, NULL, NULL)))
+        goto end;
+
+    if (!TEST_false(SSL_CTX_set1_server_cert_type(sctx, cert_type_dups, sizeof(cert_type_dups))))
+        goto end;
+
+    if (!TEST_false(SSL_CTX_set1_server_cert_type(sctx, cert_type_bad, sizeof(cert_type_bad))))
+        goto end;
+
+    if (!TEST_false(SSL_CTX_set1_server_cert_type(sctx, cert_type_extra, sizeof(cert_type_extra))))
+        goto end;
+
+    if (!TEST_false(SSL_CTX_set1_server_cert_type(sctx, cert_type_unsup, sizeof(cert_type_unsup))))
+        goto end;
+
+    if (!TEST_true(SSL_CTX_set1_server_cert_type(sctx, cert_type_just_x509, sizeof(cert_type_just_x509))))
+        goto end;
+
+    if (!TEST_true(SSL_CTX_set1_server_cert_type(sctx, cert_type_just_rpk, sizeof(cert_type_just_rpk))))
+        goto end;
+
+    ret = 1;
+ end:
+    SSL_CTX_free(sctx);
+    SSL_CTX_free(cctx);
+    return ret;
+}
+OPT_TEST_DECLARE_USAGE("certdir\n")
+
+int setup_tests(void)
+{
+    if (!test_skip_common_options()) {
+        TEST_error("Error parsing test options\n");
+        return 0;
+    }
+
+    if (!TEST_ptr(certsdir = test_get_argument(0)))
+        return 0;
+
+    rootcert = test_mk_file_path(certsdir, "rootcert.pem");
+    if (rootcert == NULL)
+        goto err;
+
+    cert = test_mk_file_path(certsdir, "servercert.pem");
+    if (cert == NULL)
+        goto err;
+
+    privkey = test_mk_file_path(certsdir, "serverkey.pem");
+    if (privkey == NULL)
+        goto err;
+
+    cert2 = test_mk_file_path(certsdir, "server-ecdsa-cert.pem");
+    if (cert2 == NULL)
+        goto err;
+
+    privkey2 = test_mk_file_path(certsdir, "server-ecdsa-key.pem");
+    if (privkey2 == NULL)
+        goto err;
+
+    cert448 = test_mk_file_path(certsdir, "server-ed448-cert.pem");
+    if (cert2 == NULL)
+        goto err;
+
+    privkey448 = test_mk_file_path(certsdir, "server-ed448-key.pem");
+    if (privkey2 == NULL)
+        goto err;
+
+    cert25519 = test_mk_file_path(certsdir, "server-ed25519-cert.pem");
+    if (cert2 == NULL)
+        goto err;
+
+    privkey25519 = test_mk_file_path(certsdir, "server-ed25519-key.pem");
+    if (privkey2 == NULL)
+        goto err;
+
+    libctx = OSSL_LIB_CTX_new();
+    if (libctx == NULL)
+        goto err;
```

Comment:
You don't seem to have loaded the null provider into the deflt ctx here. This is important: things may be working simply because they are falling back to the deflt libctx accidentally.

Reply:
It's done on line 218, for test index 15. Right now, only test 15 uses a non-default `libctx`.

Reply:
Ah! Ok. Missed that. Thanks.